syntax = "proto3";

import "google/protobuf/empty.proto";

service ChatAI {
  rpc Create(CreateRequest) returns (CreateResponse) {}
  rpc Send(SendRequest) returns (SendResponse) {}
  rpc Clear(ClearRequest) returns (google.protobuf.Empty) {}
}

// The request to create a chat
message CreateRequest {
  // The name of the user
  string name = 1;
}

// The response from creating a chat
message CreateResponse {
  string chatId = 1;
}

// The request to send a message to ChatAI
message SendRequest {
  string chatId = 1;
  string text = 2;
}

// The response from ChatAI
message SendResponse {
  string text = 1;
}

// The request to clear the chat
message ClearRequest {
  string chatId = 1;
}
